source/plugin/modularization.rst

```diff
@@ -0,0 +1,62 @@
+===================
+Code Modularization
+===================
+
+When writing Sponge plugins, you may find that you need to modularize your code. Maybe you want to split your core into
+another project so that others can make use of it. Maybe you want to utilize a service another plugin provides. Or maybe
+you just come from a country where you die if you don't modularize code (it's called Heaven). Well, with Sponge, begone
+the days of shading or shadowing entire plugins into your plugin!
+
+Problem 1: ClassNotFoundException
+=================================
+
+Let's say you want to provide some awesome serice for ``WarpService``. You took a day or two to write
```

Comment:
Awesome is in the eye of the beholder ;)   More importantly, you misspelled service.
We don't really need to know how long it took to write, just what it does (ie backing up warps to db/file). IMHO the description you provide needs a bit more reasoning.